My submission

buz-zard opened the pull request:
Referenced some basic commands in readme.

tmxyz commented on src/form/components/fields/TextField.js:
```diff
@@ -0,0 +1,49 @@
+);
+
+TextFieldComponent.propTypes = {
+  input: reduxProps.input, // eslint-disable-line
```
It is always better to disable specific rule(s) instead of disabling everything. Also, I think that you don't need to disable anything here, just add the default values to the defaultProps.

buz-zard replied:
You're right. I was pretty sure eslint didn't support imported proptypes object, but I just didn't read the error thoroughly (it clearly states that default props is needed :) ). I just need to add `.isRequired`

tmxyz commented on src/app/components/forms/LoginForm.js:
```diff
@@ -0,0 +1,75 @@
+
+
+const LoginForm = ({onSubmit}) => (
+  <form>
```
I prefer to handle the onSubmit event for the form than onClick for the submit button.

buz-zard commented on package.json:
```diff
@@ -0,0 +1,85 @@
+    "redux-form": "^7.0.3",
+    "redux-observable": "^0.16.0",
+    "redux-thunk": "^2.2.0",
+    "rxjs": "^5.4.3",
```
It's peer dependency of redux-observable - https://github.com/redux-observable/redux-observable/blob/master/package.json

buz-zard commented on src/app/state/store.js:
```diff
@@ -0,0 +1,29 @@
+  composeWithDevTools(applyMiddleware(
+    thunk,
+    routerMiddleware(history),
+    createEpicMiddleware(combineEpics(...Object.keys(epics).map(key => epics[key]))),
```
Noted.

buz-zard commented on src/resources/actions.js:
```diff
@@ -0,0 +1,25 @@
+const finishLoading = ({type, data}) => action(t.LOADING_FINISHED, {type, data});
+
+
+// async
```
I use it more like a convention/separator for actions. The actions below this comment are being handled by redux-thunk middleware. So such actions would probably need a test because they usually are a bit more complex.

buz-zard commented:
Hi Dovydas,

Thanks for thorough review, I made some changes/comments.

**1. Remove the outline from the input components**
- I could do this, but this introduces accessibility issues (while tabbing the user will not clearly see which input is focused). 
**2. The form shouldn't move when you add an error message.**
- fixed.
**3. The app doesn't work on IE11.**
- Ah yes, polyfills. I'm too used to using react-scripts. Fixed.
**4. No UI component testing.**
- I had one `withDataLoader` test, that's technically a component (hoc) test, but I added a few more component tests.
